src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -96,6 +97,23 @@ bool IsCKApplicable(const ProblemDescriptionType& problem)
         ptrs.begin(), ptrs.end(), [&args](auto& ptr) { return args.IsSupportedBy(ptr); });
 }
 
+inline void ProfilingRecordStart(const Handle& handle, HipEventPtr& start, HipEventPtr& stop)
```

Comment:
As @atamazov said, this is best handled as a class that provides `Start()`, `Stop()` and `GetTime()` and holds the `hip events` as members.